### DISCUSSION TITLE
I can tar_load and tar_read a target, but never appears in tar_make

### DISCUSSION THREAD

**Discussion by daranzolin** [ropensci/targets#1006] [Help] [answered]
See title. I can successfully call `tar_load` and `tar_read` on a target, but when I edited the function beneath it and ran `tar_make`, the target is neither run nor skipped, just never appears in the console. What are some reasons for this behavior?

Edit: the target/object does not appear in `_targets/meta`, which is most likely the problem? How do targets get written there? It appears to be wired correctly when I run `tar_glimpse`.

**Comment by wlandau** [comment]
It would depend on how you set up the pipeline and exactly how and when you are calling tar_read(). Hard to say without a reproducible example I can run to troubleshoot.

**Reply by wlandau** [reply]
For more on reprexes, you could check out the reprex package, look at https://stackoverflow.com/help/minimal-reproducible-example, or skim the other {targets} discussions that have code examples.

**Comment by wlandau** [comment, marked as answer]
It is easy to ask for a "likely" or "common" solution, but maintainers have trouble responding when there are so many possibilities, unknowns, and hidden assumptions. I wrote https://books.ropensci.org/targets/help.html to explain what I need in order to help. From https://books.ropensci.org/targets/help.html#post-the-complete-context:

> Computer code is extremely strict. In order to run properly, it needs to be exactly correct, down to the last detail. That means for every bug you discover, there are hundreds of strange and unguessable potential causes. The culprit could be a typo buried deep in the `targets` codebase, an [incorrect hidden assumption lost in translation](https://youtu.be/2ve20PVNZ18?t=52) about using `targets`, an obscure bug in an obscure R package dependency, a mistake in one of your R functions, or some other mysterious or unexpected reason. In addition, you may be the first user to experience the problem. So when you ask for help, it is not enough to state the error and ask what might be the cause. The true answer is too surprising to predict, even for the most experienced experts of R and `targets`.
